### PR TITLE
Introduce read, create, delete support for Azure Git Repos

### PR DESCRIPTION
**Pull request by KeithJRome**
## All Submissions:
-------------------------------------
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [N/A] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open PRs for the same update/change?

## What is the current behavior?
-------------------------------------
`READ`, `CREATE`, `DELETE` not supported for Azure Git Repos

Issue Number: 
 - https://github.com/microsoft/terraform-provider-azuredevops/issues/95
 - https://github.com/microsoft/terraform-provider-azuredevops/issues/94
 - https://github.com/microsoft/terraform-provider-azuredevops/issues/96

## What is the new behavior?
-------------------------------------
`READ`, `CREATE`, `DELETE` supported for Azure Git Repos; Acceptance tests run as part of CI

## Does this introduce a breaking change?
-------------------------------------
- [NO]

### CODE REVIEW

File: examples/azdo-based-cicd/main.tf [hunk 1]
```diff
@@ -79,23 +78,23 @@ resource "azuredevops_azure_git_repository" "repository" {
...
-  # ]
-
-  repository {
-    # repo_type   = azuredevops_git_repo.repository.type
-    # repo_name   = azuredevops_git_repo.repository.name
-    # branch_name = azuredevops_git_repo.repository.default_branch
-    yml_path    = "cicd/azure-pipelines-infra.yml"
-  }
-}
+# resource "azuredevops_build_definition" "cicd" {
```
Comment (TsuyoshiUshio) [inline examples/azdo-based-cicd/main.tf]: Just curious. Why don't you remove this? Why you leave as comment?

File: azuredevops/config.go [hunk 2]
```diff
@@ -68,9 +70,18 @@ func getAzdoClient(azdoPAT string, organizationURL string) (*aggregatedClient, e
 		return nil, err
 	}
 
+	// client for these APIs:
+	//	https://docs.microsoft.com/en-us/rest/api/azure/devops/git/?view=azure-devops-rest-5.1
```
[the remark this answers is not in the document]
Comment (nmiodice) [reply azuredevops/config.go]: Personally I like the quick reference to the API spec when I'm reading through a new codebase. I hope that it points somebody in the right direction who may not be familiar with the AzDO APIs.

File: examples/azdo-based-cicd/main.tf (hunk 1 above)
Comment (nmiodice) [reply examples/azdo-based-cicd/main.tf]: From the top of the file:
```
# Note: This template is *aspirational* and is included in order to exemplify the value that
# this provider will eventually provide. Based on this, keep the following in mind:
#  - Not every stanza is backed by an implementation in the provider
#  - The schema for each resource/data source will almost certainly change once implemented

# As the provider is built, the idea is that this file will slowly become uncommented and updated
# once the features are implemented.
```

File: azuredevops/config.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (KeithJRome) [reply azuredevops/config.go]: Yes, and the reference has been invaluable for debugging these provider resources

File: azuredevops/resource_azure_git_repository.go [hunk 3]
```diff
@@ -57,39 +55,51 @@ func resourceAzureGitRepository() *schema.Resource {
...
+	if err != nil {
+		return fmt.Errorf("Error creating project in Azure DevOps: %+v", err)
+	}
+
+	flattenAzureGitRepository(d, createdRepo)
 
 	return resourceAzureGitRepositoryRead(d, m)
 }
 
+// Make API call to create the repo
```
[the remark this answers is not in the document]
Comment (KeithJRome) [reply azuredevops/resource_azure_git_repository.go]: Will remove this one, you are correct - it is extraneous.

File: azuredevops/resource_project.go [hunk 4]
```diff
@@ -109,6 +111,15 @@ func waitForAsyncOperationSuccess(clients *aggregatedClient, operationRef *opera
...
 
 			if *result.Status == operations.OperationStatusValues.Succeeded {
+				// Sometimes without the sleep, the subsequent operations won't find the project...
+				delay := os.Getenv("AZDO_PRJ_CREATE_DELAY")
+				settleDelay := time.Duration(0)
+				i, err := strconv.ParseInt(delay, 10, 64)
+				if err == nil {
+					settleDelay = time.Duration(i) * time.Second
+				}
+				fmt.Printf("Creation Delay: %s\n", settleDelay.String())
```
Comment (nmiodice) [inline azuredevops/resource_project.go]:
```suggestion
				fmt.Printf("Inserting artificial delay after project creation: %s\n", settleDelay.String())
```

**Pull request merged**